doc-content/enterprise-only/springboot/creating-self-contained-image-proc.adoc

```diff
@@ -1,15 +1,28 @@
 [id='creating-self-contained-image-proc_{context}']
 = Creating a self-contained {PRODUCT} Spring Boot JAR file
 
-You can create a self-contained {PRODUCT} Spring Boot JAR file that contains multiple versions of the same KJAR file, including modules. These are files that have the same `artifactID` and `groupID` attribute values, but have different `version` values. The KJAR files are separated from the `BOOT-INF/lib` JAR files to avoid class loader collisions. Each KJAR classpath container file is isolated from other KJAR classpath container files and does not rely on the Spring Boot class loader.
+You can create a self-contained {PRODUCT} Spring Boot JAR file that contains a complete service, including the {KIE_SERVER} and one or several KJAR files. This JAR file does not depend on loading any KJAR files at runtime.
+
+If necessary, the JAR file can contain multiple versions of the same KJAR file, including modules. These are files that have the same `artifactID` and `groupID` attribute values, but have different `version` values. The KJAR files are separated from the `BOOT-INF/lib` JAR files to avoid class loader collisions. Each KJAR classpath container file is isolated from other KJAR classpath container files and does not rely on the Spring Boot class loader.
 
 .Prerequisites
 * You have an existing {PRODUCT} Spring Boot project.
+* You have completed development of one or more KJAR files for the project.
 
 .Procedure
-. Add the following properties to your Spring Boot application  `application.properties` file:
+. Build every KJAR files for the project. In the default business application, the KJAR source is contained in the `<BUSINESS-APPLICATION>/<BUSINESS-APPLICATION>-kjar` directory. Your project might include other KJAR source directories. For every KJAR source directory, complete the following steps:
```

Comment:
Build every KJAR file for... Define the variables.
.  For every KJAR source directory, complete the following steps:(make it its own step.

Reply:
It's not its own step, it's an explanation how to build every KJAR file, in case the reader does not know. Some readers do know. Would you suggest some better way to formulate here?

Reply:
This is not actually what happens. I will however do a revised version now so we have something to go with.